### PR TITLE
LPS-188969 Support UPSERT SXPBlueprint and SXPElement

### PR DESCRIPTION
**Pull request by felipelorenz**
https://liferay.atlassian.net/browse/LPS-188969
Continued from here: https://github.com/liferay-search/liferay-portal/pull/1031

### CODE REVIEW

**Comment by felipelorenz** [comment]
ci:test:relevant

**Comment by peerkar** [comment]
@felipelorenz I think this is good to go. One thing I noticed, you might still consider, is to simplify in `SXPElementResourceImpl` by combining the logic of `putSXPElement()` and  `putSXPElementByExternalReferenceCode()`.

If the first things in those methods are `sxpElement.setId(sxpElementId)`and `sxpElement.setExternalReferenceCode(externalReferenceCode)`the rest of those methods are identical. So maybe creating a new private _putSXPElement() to do the work for the both?

**Comment by felipelorenz** [comment]
ci:test:sf

**Comment by felipelorenz** [comment]
ci:test:search

**Comment by felipelorenz** [comment]
ci:test:relevant

File: modules/dxp/apps/search-experiences/search-experiences-rest-impl/src/main/java/com/liferay/search/experiences/rest/internal/resource/v1_0/SXPElementResourceImpl.java [hunk 1]
```diff
@@ -422,6 +390,23 @@ private String _getSchemaVersion() {
 		return "1.0";
 	}
 
+	private SXPElement _putSXPElement(
+			SXPElement sxpElement,
+			com.liferay.search.experiences.model.SXPElement sxpElementModel)
```
Comment (peerkar) [inline modules/dxp/apps/search-experiences/search-experiences-rest-impl/src/main/java/com/liferay/search/experiences/rest/internal/resource/v1_0/SXPElementResourceImpl.java:395]: Please rename sxpElementModel -> "serviceBuilderSXPElement"

Also Please check the casing for serviceBuilderSXPElement in the new put methods.

**Comment by felipelorenz** [comment]
ci:test:sf

**Comment by felipelorenz** [comment]
ci:test:search

**Comment by felipelorenz** [comment]
ci:test:relevant

**Comment by felipelorenz** [comment]
ci:test:sf

**Comment by felipelorenz** [comment]
ci:test:search

**Comment by felipelorenz** [comment]
ci:test:relevant

**Pull request closed without merging**